src/main/java/edu/umich/tl/CourseDelete.java

```diff
@@ -62,32 +88,193 @@ private static void checkForApiType(Properties properties) {
+		M_log.debug("getCanvasTermIdForSisTermId(): called");
+		ArrayList<Term> terms = courseForDelete.getTerms();
+		for (Term term : terms) {
+			if (term.getSisTermId().equalsIgnoreCase(sisTermIdFromPropsFile)) {
+				M_log.info("Found canvasTermId for provided SISTermId for the Term:  "+ term.getTermName());
+				return term.getCanvasTermId();
+			}
+		}
+		return null;
+	}
+	/*
+	 * Getting all the unpublished course list for a particular term from the ROOT account which is '1'. All the colleges, school of UOfM fall under ROOT.
+	 * This list is huge, canvas only provided 100 result set per api call so we need to get the rest of the courses info from the Pagination links.
+	 */
+	private static void getUnpublishedCourseList(String canvasTermIdForSisTermId, CoursesForDelete coursesForDelete, ApiCallHandler apiHandler, String url) {
+		M_log.debug("getUnpublishedCourseList(): Called");
+		HttpResponse httpResponse=null;
+		if(url!=null) {
+			//sending null for canvasTermId is fine since we get the url framed from the pagination object from the response header. 
+			httpResponse=apiHandler.getApiResponse(RequestTypeEnum.UNPUBLISHED_COURSE_LIST_PAGINATION_URL,null,url);
+		}else {
+			httpResponse = apiHandler.getApiResponse(RequestTypeEnum.UNPUBLISHED_COURSE_LIST,canvasTermIdForSisTermId,url);
+		}
+		ObjectMapper mapper = new ObjectMapper();
+		int statusCode = httpResponse.getStatusLine().getStatusCode();
+		if(statusCode!=200) {
+			M_log.error(apiCallErrorHandler(httpResponse,"Unpublished_Courses",apiHandler));
+			System.exit(0);
+		}
+		HttpEntity entity = httpResponse.getEntity();
+		List<HashMap<String, Object>> courseList=new ArrayList<HashMap<String, Object>>();
+		try {
+			String jsonResponseString = EntityUtils.toString(entity);
+			courseList = mapper.readValue(jsonResponseString,new TypeReference<List<Object>>(){});
+			for (HashMap<String, Object> course : courseList) {
+				Course aCourse=new Course()
+				.setId((Integer)course.get("id"))
+				.setCourseName((String)course.get("name"))
+				.setStartDate((String)course.get("start_at"))
+				.setEndDate((String)course.get("end_at"));
+
+				coursesForDelete.addCourse(aCourse);
+			}
+			String nextPageUrl = getNextPageUrl(httpResponse);
+			if(nextPageUrl!=null) {
+				getUnpublishedCourseList(canvasTermIdForSisTermId,coursesForDelete,apiHandler,nextPageUrl);
+			}
+		} catch (JsonParseException e1) {
+			M_log.error("JsonParseException occured getUnpublishedCourseList() : ",e1);
+		} catch(JsonMappingException e1) {
+			M_log.error("JsonMappingException occured getUnpublishedCourseList() : ",e1);
+		}  catch (IOException e) {
+			M_log.error("IOException occured getUnpublishedCourseList( ):" ,e);
 		}
 	}
+	/*
+	 * This helper method pull out the error message that is sent in case of error. Currently code distinguishes error
+	 * differently from directCanvas vs esbCanvas. This may not be always the case but ESB has Special Throttling message
```

Comment:
Note that throttling is not really an error, just a constraint that the program could accommodate.  Depending on the application it may be ok to treat it as an error.